packages/authentication/test/unit/strategy-adapter.test.ts

```diff
@@ -18,9 +22,13 @@ describe('Strategy Adapter', () => {
       // TODO: (as suggested by @bajtos) use sinon spy
       class Strategy extends MockStrategy {
         // override authenticate method to set calledFlag
-        async authenticate(req: http.ServerRequest) {
+        async authenticate(
+          this: StrategyCreated<this, this & StrategyCreatedStatic> & this,
+          req: Express.Request,
+          options?: AuthenticateOptions,
+        ) {
           calledFlag = true;
-          await super.authenticate(req);
+          await MockStrategy.prototype.authenticate.call(this, req, options);
```

Comment:
This class still extends `MockStrategy`. So why can we not do `super.authenticate(req, options)`?

Reply:
That's a good question. TL;DR: Because the typings in `@types/passport` are broken 😠 

Here is the error message reported by the compiler for the original code calling `super.authenticate`

```
The 'this' context of type 'MockStrategy' is not assignable to method's 'this' of type 'StrategyCreated<this, this & StrategyCreatedStatic> & this'.
  Type 'MockStrategy' is not assignable to type 'StrategyCreated<this, this & StrategyCreatedStatic>'.
```

As I see it, the definition of interface `StrategyCreated` is broken - it is trying to say "the original strategy object with some additional methods mixed in", but unfortunately does not convey the information that that the original strategy object remains a part of "this". Here is the definition:

```ts
    type StrategyCreated<T, O = T & StrategyCreatedStatic> = {
        [P in keyof O]: O[P];
    };
```

Reply:
As I was researching this part, I realized we can simplify our usage of StrategyCreated - see the fixup commit I added to this pull request: 94dd299